README.md

```diff
@@ -1,117 +1,149 @@
+After Puppet applies the changes described in the installation steps, you should end up with the following files in the Bolt server:
+
+- `/root/.puppetlabs/etc/bolt/bolt-defaults.yaml`
+- `/root/.puppetlabs/token`
 
-## Reference
+To test that everything was configured properly, we can run any Bolt plan that runs a PuppetDB query, for example:
 
-This section is deprecated. Instead, add reference information to your code as
-Puppet Strings comments, and then use Strings to generate a REFERENCE.md in your
-module. For details on how to add code comments and generate documentation with
-Strings, see the [Puppet Strings documentation][2] and [style guide][3].
+```
+# /root/Projects/local_plan/plans/test.pp
+
+plan local_plan::test(
+){
+  $query_results = puppetdb_query("nodes[]{}")
+  out::message("Hello world from the Bolt Server, query results: ${query_results}")
+}
+```
 
-If you aren't ready to use Strings yet, manually create a REFERENCE.md in the
-root of your module directory and list out each of your module's classes,
-defined types, facts, functions, Puppet tasks, task plans, and resource types
-and providers, along with the parameters for each.
+Run the Bolt plan:
 
-For each element (class, defined type, function, and so on), list:
+`bolt plan run local_plan::test`
 
-* The data type, if applicable.
-* A description of what the element does.
-* Valid values, if the data type doesn't make it obvious.
-* Default value, if any.
+We should see the PuppetDB query results in the terminal, and if we inspect the `puppetdb-access.log` there should be a log with a call to the local PuppetDB with a 200 Ok HTTP status:
 
-For example:
+```
+$ less /var/log/puppetlabs/puppetdb/puppetdb-access.log
 
+127.0.0.1 - - [01/Nov/2022:15:56:21 +0000] "POST /pdb/query/v4 HTTP/1.1" 200 1793 "-" "HTTPClient/1.0 (2.8.3, ruby 2.7.6 (2022-04-12))" 99 21 -
 ```
-### `pet::cat`
 
-#### Parameters
+### Running a Plan via taskplan from the Primary Server
+
+We recommend to install the [`taskplan` module](https://forge.puppet.com/modules/reidmv/taskplan) in your control-repo.
+
+- The intended use for `taskplan` is to run Plans in the Bolt server, especifically by telling it to run it on a target node, in this case, our Bolt server.
+
+**Running a Plan on the Bolt Server**
+
+This is an overview of the internal process when you offload the Plan execution from the PE Primary server to the Bolt server:
+
+1. The taskplan Task is executed to run a Plan from the Primary server targeting our Bolt server
+1. Orchestrator will connect to the PCP agent on the Bolt server and run the Task there.
+1. The Bolt server's agent will finally run the Plan on it.
+
+![bolt-server-process](diagrams/bolt-server-exec-processes.png "Bolt server execution process")
 
-##### `meow`
+**Example**
 
-Enables vocalization in your cat. Valid options: 'string'.
+From the Primary server we will run the `taskplan` task, targetting our Bolt server. Here is an example you can use:
```

Comment:
I think we should:
1. Add an example using `puppet task run` since that is probably what people will use from the PE primary CLI
2. Add a mention that the Orchestrator API call can be done from any system that has the Puppet RBAC token and has connectivity to Puppet Primary on port 8143